### ISSUE TITLE
Load fixture without a file path

### ISSUE THREAD

**Issue by alexislefebvre** [liip/LiipTestFixturesBundle#245]
> from my understanding, we always have to define our fixture in a separate class,
> what i would really like is to define the fixture data inside my test function, (maybe not always but at least sometimes)
> 
> i really like the idea of having a tight coupling between a test and the specific data it requires, to avoid creating some kind of global state that will be likely to break at scale.
> 
> Is there any way to achieve that with your lib ? something like
> 
> ```
> $this->databaseTool->loadFixturesInline(function(){ //define fixture code here... });
> ```

_Originally posted by @vulkanosaure in https://github.com/liip/LiipTestFixturesBundle/issues/64#issuecomment-1805054667_

**Comment by alexislefebvre** [comment]
@vulkanosaure `loadFixtures()` accept an array of classes: https://github.com/liip/LiipTestFixturesBundle/blob/2.x/doc/database.md#load-fixtures-

In theory, it should work if you define a class in your test file, then provide that class to the loader.

**Comment by vulkanosaure** [comment]
Thanks, that would be already better,
but would still be a class definition available to all test function.
What I would really like is the fixture definition to be local to the test function,
to forbid the possibility of another test messing up with those data and garantee a full isolation.

**Comment by vulkanosaure** [comment]
I used a horrible hack with `$GLOBALS` in the mean time, 
where the test function defines a function in `$GLOBALS` and the fixture calls it, then the test erase it

but i'd prefer to avoid something like that.

**Comment by alexislefebvre** [comment]
> Thanks, that would be already better, but would still be a class definition available to all test function. What I would really like is the fixture definition to be local to the test function, to forbid the possibility of another test messing up with those data and garantee a full isolation.

Could you please explain the use case? How could you encounter the case of “another test messing up with those data”?

**Comment by vulkanosaure** [comment]
Other test functions (from the same file) would be able to execute this
fixture, so to use the same test data.

Let's say we write another test and want to use the same fixture, but we
have new requirements so we will want to modify (manually) the data in that
fixture to fit the new test, and potentially break the previous test.

It happened to me in a previous project where I was using a single database
dump for all my tests.

**Comment by alexislefebvre** [comment]
Yes this is a common issue, when we start a project we have one fixture, then we need to test another behaviour with different data. And the project grows and more cases must be covered by tests.

You may create a fixture for every case but you'll end up with lot of duplicated code.

I would instead recommend to load the same base fixture but update it with the data you need before running the test, so that you only modify what you want.

Something like this:

```php
$entity = $this->myEntityRepository->findOneBy(['name' => 'fixture']);
$entity->setProperty('the value I need');
$this->flush();

$this->assert…
```

**Comment by vulkanosaure** [comment]
Ok that looks like an interesting idea. I was indeed thinking that I would
need at least something common, to avoid duplicating too many fixture.

I'll give this a try, thanks
